app/presenters/article_fulltext_link_presenter.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+##
+# A presenter class handle the logic to render markup for fulltext links from EDS
+class ArticleFulltextLinkPresenter
+  delegate :article_url, :article_fulltext_link_url, :link_to, :image_tag, to: :context
+  def initialize(document:, context:)
+    @document = document
+    @context = context
+  end
+
+  def links
+    return [] unless online_access_panel? || document_has_fulltext?
+    return access_panel_links.map { |link| render_fulltext_link(link) } if online_access_panel?
+    ["#{online_label} #{link_to('View on detail page', article_url(document))}"]
+  end
+
+  private
+
+  attr_reader :document, :context
+
+  def online_label
+    "<span class='online-label'>Full text</span>"
+  end
+
+  def online_access_panel?
+    document.access_panels.online?
+  end
+
+  def access_panel_links
+    document.access_panels.online.links || []
+  end
+
+  def document_has_fulltext?
+    document.html_fulltext? && !online_access_panel?
+  end
+
+  def render_fulltext_link(link)
+    if link.href == 'detail' # PDFs
```

Comment:
FYI Bill added a new attribute to `eds_fulltext_links` called `expires` which is a boolean. If true, it means we need to use our `/articles/id/type/fulltext` link (e.g., `article_fulltext_link_url`). We're not using it currently -- we'd need to add it to the SearchWorks link class.